docs/DeveloperGuide.md

```diff
@@ -281,47 +296,269 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
+      
+      Use case resumes at step 5.
+
+
+**Use case: Clearing all data**
+
+**MSS**
+1. User wants to clear all data from the current working file.
+2. User enters the clear command.
+3. RC4HDB replaces the current data file with an empty data file with the same name.
+4. RC4HDB shows a success message.
+   
+    Use case ends.
+
+**Use case: Exiting the application**
+
+**MSS**
+1. User has completed his/her tasks and wants to exit the application.
+2. User enters the exit command.
+3. RC4HDB application closes .
+   
+    Use case ends.
+
+**Extension**
+
+* 2a. User clicks on the exit button.
+  
+    * 2a1. RC4HDB application closes. 
+      
+      Use case ends.
+   
+
+**Use case: Importing data from .csv file**
+
+**MSS**
+1. User has a data file with resident’s information, and wants to view it in RC4HDB.
+2. User enters the import command, followed by the full file path of the CSV file.
+3. RC4HDB reads the file stored in the file path entered by the user.
+4. RC4HDB displays the name of the csv file after the file has been read.
+5. RC4HDB displays all the information stored in the file.
 
-  Use case ends.
+    Use case ends.
 
-* 3a. The given index is invalid.
+**Extension**
+* 1a. Information in the CSV file has not been stored in the proper format.
+  
+    * 1a1. RC4HDB shows an error message.
+      
+      Use case resumes at step 2.
 
-    * 3a1. AddressBook shows an error message.
 
+* 2a. No file could be found at the specified file path.
+  
+    * 2a1. RC4HDB shows an error message. 
+      
       Use case resumes at step 2.
 
+
+**Use case: Exporting data to .csv file**
+
+**MSS**
+1. User wants the data in a CSV file.
+2. User enters the export command, followed by the full file path and new file name.
+3. RC4HDB creates a new CSV file in the file path entered by the user.
+4. RC4HDB shows a success message.
+
+    Use case ends.
+
 *{More to be added}*
 
 ### Non-Functional Requirements
 
-1.  Should work on any _mainstream OS_ as long as it has Java `11` or above installed.
-2.  Should be able to hold up to 1000 persons without a noticeable sluggishness in performance for typical usage.
-3.  A user with above average typing speed for regular English text (i.e. not code, not system admin commands) should be able to accomplish most of the tasks faster using commands than using the mouse.
+#### Accessibility
+* The application must not require an internet connection to work
+* Inputs must be done through the Command Line Interface (CLI)
+* The application must be sufficiently light-weight to be run by older computer systems
+* Should work on any mainstream OS as long as it has Java 11 or above installed
+* A user with above average typing speed for regular English text (i.e. not code, not system admin commands) should be able to accomplish most of the tasks faster using commands than using the mouse
+
+#### Quality
+* The application must be usable by administrative staff
```

Comment:
This could potentially be a DG bug under: Not scoped clearly (i.e., hard to decide when it has been met)

I would recommend this NFR to be something along the lines of "The application must be able to be easily learnt within a week by administrative staff with limited technical knowledge."

Reply:
I agree. I think yours is more appropriate. I have adopted it and updated it. 👍